src/waltham/waltham-connection.h

```diff
@@ -382,21 +382,46 @@ wth_connection_create_registry(struct wth_connection *conn);
 struct wthp_callback *
 wth_connection_sync(struct wth_connection *conn);
 
-/** Set the callback to be called once a client requests a registry
+/** Prototype of the callback to be called whenever a client creates a
+ * registry
+ *
+ * \param registry The registry which has been created.
+ * \param user_pointer The data set in
+ * wth_connection_set_registry_callback.
+ *
+ * A function implementing this callback will receive a pointer to the
+ * newly-created registry, as well as the data previously set in
+ * wth_connection_set_registry_callback.
+ *
+ * The user_pointer data should generally identify the client this
+ * registry pertains to.
+ *
+ * \sa wth_connection_set_registry_callback
+ * \sa wth_connection_create_registry
+ *
+ * \memberof wth_connection
+ * \server_api
+ */
+typedef void (*registry_callback)(struct wthp_registry *registry, void *user_pointer);
```

Comment:
This is a public header, so the type needs to have a scoped name: `wth_registry_callback`. Or wthp? Maybe not wthp as it is not generated from XML.

Reply:
hrm, that looks like a function rather than a type. `wth_registry_callback_func`?